submissions/api.py

```diff
@@ -467,8 +469,8 @@ def get_top_submissions(course_id, item_id, item_type, number_of_top_scores, use
             for score in scores
         ]
 
-    # Always store the retrieved list in the cache
-    cache.set(cache_key, top_submissions, TOP_SUBMISSIONS_CACHE_TIMEOUT)
+        # Always store the retrieved list in the cache
+        cache.set(cache_key, top_submissions, TOP_SUBMISSIONS_CACHE_TIMEOUT)
```

Comment:
It makes sense to me to set it after the database call. We check on line 440 if the top submissions are cached yet, if not, we query, then set the results in the cache. What did you have in mind?